_posts/2021-09-29-boa-release-13.md

```diff
@@ -0,0 +1,101 @@
+
+```rust
+let mut context = Context::new();
+let add_value = 1.0;
+context.register_global_closure("addOne", 1, move |_, args, _| {
+    let argument = args.get(0).unwrap().as_number().unwrap();
+    Ok(JsValue::new(argument + add_value))
+})?;
+assert_eq!(context.eval("addOne(41)")?, 42.into());
+```
+
+### Boa prelude
+
+This release exposes a prelude of useful things that are already imported. This should make it more ergonomic to work with.
+
+```rust
+use boa::prelude::*; // This would import all the commonly-used things
+
+fn main() {
+	let number = JSValue::number(3.1415);
+	let realm = Realm::create();
+	let mut engine = Interpreter::new(realm);
+	let result = forward(&mut engine, "Math.PI");
+}
+```
+
+To see the full capabilities of this feature, take a look at our [examples](https://github.com/boa-dev/boa/blob/master/boa/examples/closures.rs). Thanks to @HalidOdat and @jedel1043 for their work on this.
+
+## Test 262
+
+Test262 is the implementation conformance test suite maintained by TC39. It's used by nearly all engines to measure how conformant they are to the specification. Boa pulls the tests in-tree and runs them against all PRs.
+
+Since v0.12 we have managed to pass 6391 more tests and are 7% more conformant. This has been great progress by all involved and we hope this continues to improve. Most of these new passes have come from refactors across the codebase which have had little impact on performance, in fact, v0.13 is much faster than v0.12.
+
+You can track Boa's conformance to the specification [here](https://boa-dev.github.io/boa/test262/#)
+
+## Keeping up with Rust
+
+As we work on Boa Rust is changing underneath us. This means we need to make sure our code is conforming the latest standards and we're taking advantage of the best optimizations Rust can provide. In order to do this we rely on Clippy, this is Rust's in-house linter. We updated our code to [respect the latest Rust version](https://github.com/boa-dev/boa/pull/1352) and updated [formatting](https://github.com/boa-dev/boa/pull/1356). Often a nice bonus of doing this is getting performance improvements for free as Clippy gets more "smarter". Thanks to @neeldug and @RageKnify for the work in this area.
+
+## Keeping things in order
+
+Previously object properties were stored using [`FXHash`](https://crates.io/crates/fxhash). Despite being very fast it didn't offer any guarantees about the order. Properties need to retain the same order they were entered. In order to achieve this we switched over to [IndexMap](https://docs.rs/indexmap/1.7.0/indexmap/). Traits in Rust make this easy as IndexMap was designed to be a drop-in replacement for other HashMap implementations by following the same Trait. Performance losses were minimal on some benchmarks and we actually made some gains in others. Thanks to @raskad for their work on this
```

Comment:
I think my change to this section was lost in the last commit.

```suggestion
Previously object properties were stored using [`FxHashMap`](https://docs.rs/rustc-hash/1.1.0/rustc_hash/type.FxHashMap.html). Despite being very fast it didn't offer any guarantees about the order. Properties need to retain the same order they were entered. In order to achieve this we switched over to [IndexMap](https://docs.rs/indexmap/1.7.0/indexmap/). Traits in Rust make this easy as IndexMap was designed to be a drop-in replacement for other HashMap implementations by following the same Trait. Thanks to IndexMap offering the possibility to use alternative hashing algorithms, we could continue using the fast [`FxHasher`](https://docs.rs/rustc-hash/1.1.0/rustc_hash/struct.FxHasher.html) algorithm from rustc. Performance losses were minimal on some benchmarks and we actually made some gains in others. Thanks to @raskad for their work on this
```

Reply:
done